spyder/plugins/completion/plugin.py

```diff
@@ -406,6 +398,8 @@ def on_close(self, cancelable=False) -> bool:
                 can_close |= provider_can_close
                 if provider_can_close:
                     provider.shutdown()
+            else:
+                can_close |= True
```

Comment:
Same remark as for the function `can_close()` above.

Reply:
I think the same applies for this, so indeed, the method should return `True` only if all the providers can be closed and return `False` otherwise :+1: